flagd/pkg/service/flag-sync/handler.go

```diff
@@ -49,19 +49,7 @@ func (s syncHandler) SyncFlags(req *syncv1.SyncFlagsRequest, server syncv1grpc.F
 		defer cancel()
 	}
 
-	flags, _, err := s.store.GetAll(ctx, &selector, watcher)
-	if err != nil {
-		return status.Error(codes.Internal, "error retrieving flags from store")
-	}
-	flagsString, err := json.Marshal(flags)
-	if err != nil {
-		return status.Error(codes.DataLoss, "error marshalling flags")
-	}
-	err = server.Send(&syncv1.SyncFlagsResponse{FlagConfiguration: string(flagsString), SyncContext: syncContext})
-
-	if err != nil {
-		return err
-	}
```

Comment:
`Watch` makes this much cleaner; we don't need to call `GetAll` for an initial payload.